crates/backend/src/ast.rs

```diff
@@ -379,6 +382,44 @@ pub struct Variant {
     pub comments: Vec<String>,
 }
 
+/// Information about a Trait being exported
+#[cfg_attr(feature = "extra-traits", derive(Debug, PartialEq, Eq))]
+#[derive(Clone)]
+pub struct Trait {
+    /// The name of the trait in Rust code
+    pub rust_name: Ident,
+    /// The name of the trait in JS code
+    pub js_name: String,
+    /// All the methods of this trait to export
+    pub methods: Vec<TraitMethod>,
+    /// The doc comments on this trait, if provided
+    pub comments: Vec<String>,
+    /// Whether to generate a typescript definition for this trait
+    pub generate_typescript: bool,
```

Comment:
is there any point to have this field? traits/interfaces only exist in TS

Reply:
Sorry i never answered this. You can use traits without typescript definitions. The proposed way of doing this was making the interface contain ``Symbol``s for each method. So when you want to run the function of a class implementing such trait you do
```js
struct[Trait.someMethodName]()
```

This is still the same in Typescript, because when you add a method of the same name on a class on Typescript, you override the method definition of the interface. Which is not normal behavior on rust